### DISCUSSION TITLE
Trackpad not moving

### DISCUSSION THREAD

**Discussion by joek** [george-norton/peacock#5] [Q&A] [answered]
Hi everybody, 
I just received the PCB from JLPCB and flashed the latest firmware (Option 1). The rotary encoder as well as the buttons are working fine.

If I try to use the trackpad nothing happens. It seems like it is not even connected. I tried already different VIK Cables as well as turning it around. 

One thing I came across, do I need to stet the jumper JP1?

Also what is the best way to enable debugging?

**Comment by george-norton** [comment]
What sort of surface have you applied to the sensor?

**Comment by george-norton** [comment]
If you don't have a surface, I would expect no movement. The cheapest and easiest surface is adhesive vinyl, but you will need 3 layers for reliable tracking. Other surfaces may need some tweaks to tune the sensor.

**Reply by joek** [reply]
So far I haven’t glued the surface to the trackpad. I used some pla plastic or wood as test layer. But I could not even spot a tiny movement.

**Comment by joek** [comment]
I can get som debug messages from the button press but no messages from the trackpad. But I am certain I am missing some config.
<img width="514" alt="Screenshot 2024-06-25 at 07 52 51" src="https://github.com/george-norton/peacock/assets/202380/292bb229-fcb3-4c59-87d3-a98f9c2bc265">

See current config: https://github.com/joek/qmk_firmware-peacock-v1/tree/peacock

**Comment by george-norton** [comment]
Which release of the PCBs did you get fabbed? Also are you building the v1_00 or v1_01 firmware target?

Some simple experiments you can try:
- If you touch the back of the sensor PCB do you get any movement (in this case the FR4 acts like a surface).
- If you press the two right most buttons together, does the LED on the sensor board toggle?

________________________________

**Reply by joek** [reply]
Ah, that is a good question. I was using the peacock branch as I am using a Mac and thought starting with the Option 1 might ease some things. When I am home I will try the V1_01 firmware..

**Reply by george-norton** [reply, marked as answer]
I made a mistake on the first run of boards so v1.00 doesn't have a functional I2C interface - it requires 2 bodge wires to make it work. The hardware was fixed before I announced this project, so I hope the 10 boards I ordered are the only v1.00 ones out there. Everyone else should be on the fixed v1.01 firmware.

**Reply by george-norton** [reply]
If you are using a wood or PLA surface it is very likely you will need to adjust the sensor tuning. Uncommenting this line will almost certainly give you better results than starting with the default Vinyl tuning:

https://github.com/george-norton/qmk_firmware/blob/cb74d913a63dcf1946b9db0efc953cc9e45bb517/keyboards/georgen/peacock/keymaps/default/config.h#L2

Looking forward to seeing what you make!

**Reply by joek** [reply]
Perfect, trackpad is doing some movement. Many thanks, a big step in the right direction. 

Next steps:
- Apply proper vinyl surface
- Check functionality on Macos
- Work on a custom case

**Reply by george-norton** [reply]
Great! On MacOS you will get limited functionality. Apple do not support 3rd party trackpads, so the device will fallback to reporting as a mouse. Some basic gestures will work (tapping, tap and hold, two finger scroll).
